File: shoebill_plugins/JobPlugin/src/main/java/lt/ltrp/FactionAcceptCommands.java
package lt.ltrp;

import lt.ltrp.data.Color;
import lt.ltrp.data.PlayerJobData;
import lt.ltrp.object.LtrpPlayer;
import lt.ltrp.object.Rank;
import net.gtaun.shoebill.common.command.Command;
import net.gtaun.shoebill.object.Player;

/**
 * @author Bebras
 *         2016.03.04.
 */
public class FactionAcceptCommands {

    private JobPlugin jobs;

    public FactionAcceptCommands(JobPlugin jobPlugin) {
        this.jobs = jobPlugin;
    }

    @Command
    public boolean faction(Player p) {
        LtrpPlayer player = LtrpPlayer.get(p);
        FactionInviteOffer offer = player.getOffer(FactionInviteOffer.class);
        if(offer == null) {
            player.sendErrorMessage("Jums niekas nesiûlo prisijungti prie frakcijos.");
        } else if(offer.isExpired()) {
            player.sendErrorMessage("Pasiûlymo laikas pasibaigë.");
        } else {
            player.getOffers().remove(offer);
            LtrpPlayer leader = offer.getOfferedBy();
            PlayerJobData leaderData = JobController.get().getJobData(leader);
            Rank minRank = leaderData.getJob().getRanks().stream().min((r1, r2) -> Integer.compare(r1.getNumber(), r2.getNumber())).get();
            jobs.setJob(player, leaderData.getJob(), minRank);
            player.sendMessage(Color.NEWS, "Prisijungëte prie frakcijos \"" + leaderData.getJob().getName() + "\". Jûsø rangas: " + minRank.getName());
            leader.sendMessage(Color.NEWS, player.getCharName() + " prisijungë prie jûsø frakcijos, jam paskirtas rangas " + minRank.getName());
        }
        return true;
    }

}

File: shoebill_plugins/player-job/src/main/java/lt/ltrp/player/job/command/FactionAcceptCommands.kt
package lt.ltrp.player.job.command

import lt.ltrp.data.Color
import lt.ltrp.player.job.data.FactionInviteOffer
import lt.ltrp.player.`object`.LtrpPlayer
import lt.ltrp.player.job.PlayerJobController
import net.gtaun.shoebill.common.command.Command
import net.gtaun.shoebill.`object`.Player

/**
 * @author Bebras
 *         2016.03.04.
 */
class FactionAcceptCommands {

    @Command
    fun faction(p: Player): Boolean {
        val player = LtrpPlayer.get(p)
        val offer = player.getOffer(FactionInviteOffer::class.java)
        if(offer == null) {
            player.sendErrorMessage("Jums niekas nesiûlo prisijungti prie frakcijos.");
        } else if(offer.isExpired) {
            player.sendErrorMessage("Pasiûlymo laikas pasibaigë.");
        } else {
            player.offers.remove(offer)
            val leader = offer.offeredBy
            val job = leader.jobData?.job
            if(job != null) {
                PlayerJobController.instance.setJob(player, job)
                player.sendMessage(Color.NEWS, "Prisijungëte prie frakcijos \"" + leader.jobData?.job?.name + "\". Jûsø rangas: " + player.jobData?.rank?.name)
                leader.sendMessage(Color.NEWS, player.charName + " prisijungë prie jûsø frakcijos, jam paskirtas rangas " + player.jobData?.rank?.name)
            }
        }
        return true
    }

}
